<launch>
  <!-- Image topic to subscribe to -->
  <arg name="image_topic" default="/syn_interface/camera_sensor_1"/>
  <!-- Cone topic to publish to -->
  <arg name="cone_topic" default="/perception/cone_image_map"/>
  <!-- Use simulated sensors -->
  <arg name="simulating" default="false"/>
  <!-- Use bag file -->
  <arg name="bag_file" default="false"/>
  <!-- Display opencv window -->
  <arg name="display" default="false"/>
  <!-- Include image detection launch file -->
  <!-- <include file="$(find perception_meta)/launch/camera_perception.launch"> -->
    <!-- <arg name="simulating" value="$(arg simulating)"/> -->
    <!-- <arg name="image_topic" value="$(arg image_topic)"/> -->
    <!-- <arg name="bag_file" value="$(arg bag_file)"/> -->
  <!-- </include> -->
  <group ns="perception">
    <group ns="opencv">
      <!-- Launch thresholding node -->
      <node pkg="thresholding" type="thresholding" name="thresholding" output="screen" required="true">
        <param name="image_topic" value="$(arg image_topic)"/>
        <param name="cone_topic" type="str" value="$(arg cone_topic)"/>
        <param name="simulating" type="bool" value="$(arg simulating)"/>
        <param name="display" type="bool" value="$(arg display)"/>
      </node>
    </group>
  </group>

  <group ns="perception">
    <group ns="opencv">
      <!-- Launch thresholding node -->
      <node pkg="thresholding" type="thresholding" name="thresholding2" output="screen" required="true">
        <param name="image_topic" value="/syn_interface/camera_sensor_2"/>
        <param name="cone_topic" type="str" value="$(arg cone_topic)"/>
        <param name="simulating" type="bool" value="$(arg simulating)"/>
        <param name="display" type="bool" value="$(arg display)"/>
      </node>
    </group>
  </group>

  <node pkg="syn_interface" type="syn_interface" name="syn_interface" output="screen" required="true"/>
</launch>
